tests/foreman/cli/test_organization.py

```diff
@@ -444,60 +305,159 @@ def test_positive_add_user_by_id(self):
         org = make_org()
         user = make_user()
         Org.add_user({
-            'name': org['name'],
+            'id': org['id'],
             'user-id': user['id'],
         })
+        org = Org.info({'id': org['id']})
+        self.assertIn(user['login'], org['users'])
 
     @tier2
     def test_positive_remove_user_by_id(self):
-        """@Test: Check if a User can be removed from an Org
+        """Add an user to organization and remove it by its ID
 
         @Feature: Organization
 
-        @Assert: User is removed from the org
+        @Assert: The user is added then removed from the organization
         """
         org = make_org()
         user = make_user()
         Org.add_user({
-            'name': org['name'],
+            'id': org['id'],
             'user-id': user['id'],
         })
-
         Org.remove_user({
-            'name': org['name'],
+            'id': org['id'],
             'user-id': user['id'],
         })
+        org = Org.info({'id': org['id']})
+        self.assertNotIn(user['login'], org['users'])
 
     @tier2
-    @stubbed()
     def test_positive_remove_user_by_name(self):
-        """@test: Create different types of users then add/remove user
-        by using the organization name
+        """Add an user to organization and remove it by its login and
+        organization name
 
         @feature: Organization
 
         @assert: The user is added then removed from the organization
+        """
+        org = make_org()
+        user = make_user()
+        Org.add_user({
+            'name': org['name'],
+            'user': user['login'],
+        })
```

Comment:
Are you sure it's a good idea? Three points:
- This is `test_positive_remove_***` test, we're verifying if removing works ok, not addition. For addition we have `test_positive_add_***` tests.
- `return_code` is still indirectly verified, so if addition was failed (which is not the object of testing in `test_positive_remove_***` test, btw), test will fail anyway
- If i add those kind of assertions, `test_positive_remove_***` will duplicate 100% of `test_positive_add_***` code, making `test_positive_add_***` tests redundant.

Please share your thoughts if you have another vision.